packages/react-components/react-combobox/src/components/Combobox/renderCombobox.tsx

```diff
@@ -18,7 +18,7 @@ export const renderCombobox_unstable = (state: ComboboxState, contextValues: Com
       <ComboboxContext.Provider value={contextValues.combobox}>
         <slots.input {...slotProps.input} />
         {slots.expandIcon && <slots.expandIcon {...slotProps.expandIcon} />}
-        {state.open ? popup : null}
+        {state.open || state.hasFocus ? popup : null}
```

Comment:
I'm a bit wary of making the listbox slot nullable since I don't think the author-facing API should be determined by what makes the internal logic easier. On the other hand, I think it's probably fine to allow authors to set `listbox={null}`. Are you OK if I implement that in a separate PR?

Reply:
I did update `useComboboxStyles` to only check for `open` though, that makes much more sense 👍